geopandas/base.py

```diff
@@ -288,9 +291,8 @@ def total_bounds(self):
 
     @property
     def sindex(self):
-        if not self._sindex_valid:
+        if not self._sindex:
             self._generate_sindex()
-            self._sindex_valid = True
```

Comment:
i c, was mislead by the name...

I propose change it to `_sindex_generated` and move it into the `_generate_sindex` function